packages/react-data-grid/src/RowsContainer.js

```diff
@@ -1,35 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const SimpleRowsContainer = (props) => {
-  return (
-    <div key="rows-container">
-      {props.rows}
-    </div>
-  );
-};
+export const DEFAULT_CONTEXT_MENU_ID = 'rgdContextMenu';
+
+const SimpleRowsContainer = (props) => <div key="rows-container">{props.rows}</div>;
 
 SimpleRowsContainer.propTypes = {
   width: PropTypes.number,
   rows: PropTypes.array
 };
 
+export const getNewContextMenuProps = ({ contextMenuId, rowIdx, idx }) => ({
+  rowIdx, idx, id: contextMenuId || DEFAULT_CONTEXT_MENU_ID
+});
+
 class RowsContainer extends React.Component {
   constructor(props) {
     super(props);
     this.plugins = props.window ? props.window.ReactDataGridPlugins : window.ReactDataGridPlugins;
```

Comment:
This is an old code but is there a use case of `props.window`? I wonder if anyone is actually using it, I do not think it can even be populated. This can be cleaned up (probably in a separate PR)

https://github.com/adazzle/react-data-grid/blob/master/packages/react-data-grid/src/Canvas.js#L284

Reply:
@amanmahajan7 can it be server side rendering?
I think this was added back in the days due to IE8 support problems, having the context menu added directly would break IE8 apart

Reply:
I am not sure but that does seem like a workaround for SSR. We can revisit this code once RDG officially supports server side rendering.